Sources/TSCBasic/Lock.swift

```diff
@@ -178,12 +178,22 @@ public final class FileLock {
             throw FileSystemError(.notDirectory, lockFilesDirectory)
         }
         // use the parent path to generate unique filename in temp
-        var lockFileName = (resolveSymlinks(fileToLock.parentDirectory).appending(component: fileToLock.basename)).components.joined(separator: "_")
+        let pathComponents = (resolveSymlinks(fileToLock.parentDirectory).appending(component: fileToLock.basename)).components
+        var lockFileName = pathComponents.joined(separator: "_") + ".lock"
```

Comment:
I think we don't need this part, right? At least I don't see any uses of `pathComponents` beyond the single one.

Reply:
Also, I added the ".lock" at the end so that it's part of the length calculation, and I didn't want that to get missed by people reading the code (after the change has been made).  So splitting out seemed a little clearer rather than a very long line.

Reply:
I don't have a strong preference, just pointing out that it isn't necessarily needed in case splitting it up was a leftover from a previous version of the code.

Reply:
Right, I see.  To make this clear, I'll move it back together again.  Thanks!